### PR TITLE
Nested .get()

### LINKED ISSUE #767: Nested .get()
It would be mighty handy to be able to `.get()` more than one level down. So instead of:

```js
promiseReturningFunc()
  .get('a')
  .get('b')
  .get('c')
  .get('d')
  .then(doSomethingToD)
```

you could use:

```js
promiseReturningFunc()
  .get(['a', 'b', 'c', 'd'])
  .then(doSomethingToD)
```

or even:

```js
promiseReturningFunc()
  .get('a.b.c.d')
  .then(doSomethingToD)
```

like [lodash's _.get()](https://lodash.com/docs#get).

Thanks for making such an awesome library.

### PR DESCRIPTION
**Pull request by dsernst**
Closes #767

This PR adds functionality, documentation, and a unit test for this sort of behavior:

```js
promiseReturningFunc()
  .get(['a', 'b', 'c', 'd'])
  .then(doSomethingToD)
```

### CODE REVIEW

**Comment by dsernst** [comment]
This is the only Travis error, appears unrelated:

> 26 Oct 02:48:33 - GET https://saucelabs.com/rest/v1/undefined/tunnels?full=1: HTTP response code indicated failure.26 Oct 02:48:33 - Response: {"error": "Not authorized"}.26 Oct 02:48:33 - Error shutting down overlapping tunnels.26 Oct 02:48:33 - Sauce Connect could not establish a connection.26 Oct 02:48:33 - Please check your firewall and proxy settings.

File: src/call_get.js [hunk 1]
```diff
@@ -106,6 +106,13 @@ function indexedGetter(obj) {
     return obj[index];
 }
 Promise.prototype.get = function (propertyName) {
+    if (util.isArray(propertyName)) {
+        if (propertyName.length > 1) {
+            var firstProperty = propertyName.shift();
+            return this.get(firstProperty).get(propertyName);
```
Comment (benjamingr) [inline src/call_get.js]: There should probably be a fast path for where the property is a non-promise.

**Comment by petkaantonov** [comment]
You should probably create a third kind of getter "arrayGetter" and use that as a third case in the if else chain. It will be same amount of code but far more efficient

**Comment by petkaantonov** [comment]
Also, wouldn't multiple args be more comfortable to use than having to create an array:

```js
promiseReturningFunc().get('a', 'b', 'c', 'd')
```

**Comment by bergus** [comment]
> wouldn't multiple args be more comfortable to use than having to create an array

More comfortable for handwritten code, yes, but not when the "path" is dynamically constructed. Imo both should be allowed.
```js
if (arguments.length != 1 || typeof arguments[0] == "object") {
    var path = [].concat.apply([], arguments);
    … // use arrayGetter
} else
    … // fast path for single getter
}
```

**Comment by petkaantonov** [comment]
master was replaced with 3.0, which is divergent from 2.x

**Comment by petkaantonov** [comment]
@bergus I don't think dynamic path is so common, I don't recall ever needing it in application code and in utility code I have only needed dynamic path to implement an utility that parses a string like `a.b.c.d` to a function that reads the property path `a.b.c.d` from an object

**Comment by petkaantonov** [comment]
closing 2.0 pr, please retarget to 2.x branch if still applicable

**Pull request closed without merging**